### PR TITLE
ITNFA-11: added registration with the national fingerprint server

### PR DESCRIPTION
**Pull request by pawelcieszko**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public void register(NationalBioServerClient client, BiometricSubject subject, M2SysCaptureResponse capture) {
+        try {
+            String response = client.enroll(subject.getSubjectId(), capture.getTemplateData());
+            M2SysResults results = XmlResultUtil.parse(response);
+
+            if (!results.isRegisterSuccess()) {
+                String responseValue = results.firstValue();
+                LOG.info("Got error response from the national server: {}. Checking if tied to patient.", responseValue);
+                Patient patient = patientHelper.findByNationalFpId(responseValue);
+                handleRegistrationError(subject, results, patient, client);
```
Comment (pawelcieszko) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: am I supposed to handle error with orphan fingerprint ID fixing on national server as well?
method handleRegistrationError is called while local and national registration

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/RegistrationService.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.openmrs.module.m2sysbiometrics.service;
+
+import org.openmrs.module.m2sysbiometrics.bioplugin.LocalBioServerClient;
+import org.openmrs.module.m2sysbiometrics.bioplugin.NationalBioServerClient;
+import org.openmrs.module.m2sysbiometrics.model.M2SysCaptureResponse;
+import org.openmrs.module.registrationcore.api.biometrics.model.BiometricSubject;
+
+public interface RegistrationService {
+
+    void register(LocalBioServerClient client, BiometricSubject subject, M2SysCaptureResponse capture);
```
Comment (pgesek) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/service/RegistrationService.java]: can't we just inject the servers to the bean and remove them from the interface methods?

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java (hunk 1 above)
Comment (pgesek) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: with the national registration, we do not want to reuse the error handling that changes subject ids etc

Comment (pgesek) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: Just drop the error handling here, and for now just log the error

**Review by pgesek — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java (hunk 1 above)
Comment (pawelcieszko) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: ok I will do it

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/RegistrationService.java (hunk 2 above)
Comment (pawelcieszko) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/service/RegistrationService.java]: ok I will do it

**Review by pgesek — APPROVED** [review]

**Pull request merged**